### DISCUSSION TITLE
Module that detects locale (header) and redirects to the right storeview

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#3838] [Questions & Answers]
Afaik the advised way to implement multiple languages in magento is to create a a storeview for each language, and set the locale for that storeview. 

Is there some module that can check on the Accept-Language header (or geolocation) and automatically redirect to the correct storeview? 

Accept-Language
https://manytools.org/http-html-text/browser-language/

**Comment by pquerner** [comment]
Yea, you can do that in Magento using Observers. However, maybe its possible for you to do that in your Nginx configs, as they are much faster and doesnt have to parse each request (slower PHP code).
